src/cdk/coercion/string-array.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/**
+ * Coerces a value to an array of trimmed non-empty strings.
+ * Useful for defining CSS classes or table columns.
+ */
+export function coerceStringArray(value: any, separator: string | RegExp = ' '): string[] {
+    if (value == null) {
+        return [];
+    }
+    const stringArray = Array.isArray(value) ?
+        value.map(item => `${item}`) :
+        value.toString().split(separator);
+    return stringArray
+        .map((item: string) => item.trim())
+        .filter((item: string) => item.length > 0);
+}
```

Comment:
Here's my proposed edit
```ts
export function coerceStringArray(value: any, separator: string | RegExp = ' '): string[] {
  const result = [];

  if (value != null) {
    const sourceValues = Array.isArray(value) ? value : `${value}`.split(separator);
    for (const sourceValue of sourceValues) {
      const trimmedString = `${sourceValue}`.trim();
      if (trimmedString) {
        result.push(trimmedString);
      }
    }
  }

  return result;
}
```
This only loops through the values once, avoids the overhead of extra function calls, and uses the truthiness of each subvalue to gate adding to the results.

Reply:
I agree. That implementation is better.